Feature/jwt refresh token, change password, 사용자 정보 조회

shawnchoi8 opened the pull request:
## Summary (요약)
refresh token 구현, 비밀번호 변경 기능 구현, 현재 로그인한 사용자 정보 조회

## Changes (변경 사항)
- refreshToken을 이용해 새로운 accessToken 발급 : POST /api/auth/refresh
- 현재 로그인한 사용자 정보 조회 : GET /api/users/me
- 비밀번호 변경 : PUT /api/users/password


## Type of change
- [ ] Bug fix (버그 수정)
- [x] New feature (새로운 기능 추가)
- [ ] Refactoring (리팩토링)
- [ ] Documentation (문서화)
- [ ] Tests (테스트 추가/수정)

---

## 📌 PR Checklist
- [x] 브랜치 이름 규칙 확인 (feature/*, fix/* 등)
- [x] PR 대상 브랜치 확인 (develop)
- [x] 최신 develop 반영 완료 (merge or rebase)
- [x] 코드 자동 정렬 실행 (⌥⌘L / Ctrl+Alt+L) & 불필요한 import 제거
- [x] 로컬 빌드/테스트 확인 완료

shawnchoi8 committed:
chore: add basic JWT configuration to application.yaml

shawnchoi8 committed:
feat: extend JwtUtil with refresh token generation and validation, set expirations (30m access, 24h refresh)

shawnchoi8 committed:
feat: add refresh token fields to LoginResponseDto

shawnchoi8 committed:
feat: generate accessToken, refreshToken when login for response

shawnchoi8 committed:
feat: Dtos for RefreshToken

shawnchoi8 committed:
feat: Add JWT refresh token support

- Implement access token (30min) and refresh token (24h) system
- Add token renewal API endpoint POST /api/auth/refresh
- Update login response to include both token types
- Support token type validation in JwtUtil
- Create AuthController for authentication-related endpoints

shawnchoi8 committed:
fix: add missing refresh token expiration and return value in login response

- Added default refresh token expiration (86400000 ms) to JwtUtil
- Renamed expiration fields in LoginResponseDto for clarity
- Fixed UserService to include refreshToken in login response builder

shawnchoi8 committed:
Merge branch 'develop' into feature/jwt-refresh-token

shawnchoi8 committed:
feat: Implement user profile endpoint

- Add GET /api/users/me to retrieve current user information
- Extract user data from JWT token authentication
- Return comprehensive user profile including role and status

shawnchoi8 committed:
feat: Add password change endpoint

- Implement password change with current password verification
- Add validation for password confirmation matching
- Prevent reusing current password
- Add proper error handling with 400 status codes
- Add PUT /api/users/password endpoint

soojjung commented on src/main/resources/application.yaml:
```diff
@@ -33,8 +33,14 @@ springdoc:
     org.hibernate.orm.jdbc.bind: trace   # Spring Boot 3.x에서 바인딩 값 로그 찍기
+
+jwt:
+  secret: your-very-secure-secret-key-here-make-it-long-and-random
```
uuid로 랜덤값 생성 부탁드립니다.

shawnchoi8 replied:
이 부분은 배포 시점에 github secret key를 활용하여 refactor 하도록 하겠습니다

soojjung commented on src/main/java/com/iherbyou/user/dto/ChangePasswordRequestDto.java:
```diff
@@ -0,0 +1,24 @@
+    private String newPassword;
+
+    @NotBlank(message = "비밀번호 확인은 필수입니다.")
+    private String confirmPassword;
```
confirmPassword는 프론트에서 처리하고 넘기지 않는거로 하겠습니다~

shawnchoi8 replied:
네 수정 완료했습니다

soojjung replied:
Good

shawnchoi8 committed:
refactor: Move password confirmation validation to frontend

- Remove confirmPassword from backend DTO and validation logic
- Backend now focuses on essential security checks only

shawnchoi8 committed:
fix: password maximum length setting (x -> 20)

shawnchoi8 committed:
chore: remove todo

soojjung reviewed (approved):
수고하셨습니다 ⚽️

soojjung reviewed (commented):
혹시 refresh 엔드포인트가 `/api/auth/refresh` 인가요? `users`가 아니라?
